File: ws/bonifaz/u1/w08CDtoJava/ChickenFarmSimulator/src/ec/edu/espe/farmsimulator/controller/SaveFile.java
package ec.edu.espe.farmsimulator.controller;

import ec.edu.espe.farmsimulator.model.Chicken;
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Date;
import java.util.Scanner;

/**
 *
 * @author Christian Bonifaz, Code Creators, DCCO-ESPE
 */

public class SaveFile {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("C:\\Users\\mateo\\Downloads\\ChickensData.txt"))) {
            writer.write("");
            System.out.println("Christian's Chicken farm simulator V0.6"); 
            System.out.println("\nChicken Data");
   
            System.out.print("ID: ");
            int id = scanner.nextInt();
                    
            System.out.print("Name: ");
            String name = scanner.next();

            System.out.print("Color: ");
            String color = scanner.next();

            System.out.print("Age : ");
            int age = scanner.nextInt();

            System.out.print("Molting [true/false]: ");
            boolean molting = scanner.nextBoolean();

            Date bornOnDate=new Date();
            Chicken chicken = new Chicken(id, name, color, age, molting, bornOnDate);
            System.out.println("\n--- Saved Data ---");
            System.out.println(chicken);
            writer.write(chicken.toString());
            writer.newLine();
            }  
            catch (IOException e) {
            } 
    }
}

File: hw/bonifaz/u1/hw09CDToJavaScrip/ChickenFarmSimulator/SaveFile.js
const fs = require('fs');
const readline = require('readline');
const Chicken = require('./Chicken');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});


console.log("\nChristian's Chicken farm simulator V0.6");
console.log("Chicken Data\n");

const questions = ["ID: ", "Name: ", "Color: ", "Age: ", "Molting [true/false]: "];

let chicken;
const askQuestion = (numberQuestion, answers = []) => {
    rl.question(questions[numberQuestion], (answer) => {
        answers.push(answer);
        if (numberQuestion === questions.length - 1) {
            rl.close();
            createChicken(answers);
        } else {
            askQuestion(numberQuestion + 1, answers);
        }
    });
};

const createChicken = ([id, name, color, age, molting]) => {
    const bornOnDate = new Date();
    chicken = new Chicken(id, name, color, age, molting === 'true', bornOnDate);
    saveChickenToFile(chicken);
};

const saveChickenToFile = (chicken) => {
    const filePath = 'C:\\Users\\mateo\\Downloads\\ChickensData.txt';
    fs.writeFileSync(filePath, chicken.toString() + '\n');
    console.log("\n--- Saved Data ---\n");
};

askQuestion(0);
